libs/util/httpRequest.js

```diff
@@ -121,15 +121,19 @@ function httpRequest(options, attempt) {
             options.retry.interval *
             Math.pow(2, currentAttempt);
 
+        controller = new AbortController();
+        currentAttempt += 1;
         return delayPromise(function () {
-            return httpRequest(options, currentAttempt + 1);
+            return io(options, controller).catch(retry);
```

Comment:
@redonkulus the important change is: `httpRequest` became a closure instead of a recursive function. However, note that the `catch` handler is recursive (and it's only possible since we named the handler). `retry` updates the `currentAttempt` and the `controller` and then triggers a new `io` call.

Recursive promises are dangerous since they can block the loop for an while, but since `delayPromise` calls `setTimeout` internally, we are safe here.

Reply:
would be good to add description as a comment in the code